engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/SubWorkflowExecutionActor.scala

```diff
@@ -0,0 +1,265 @@
+    case Event(Execute, _) =>
+      if (restarting) {
+        subWorkflowStoreActor ! QuerySubWorkflow(parentWorkflow.id, key)
+        goto(SubWorkflowCheckingStoreState)
+      } else {
+        prepareSubWorkflow(createSubWorkflowId())
+      }
+  }
+  
+  when(SubWorkflowCheckingStoreState) {
+    case Event(SubWorkflowFound(entry), _) => 
+      prepareSubWorkflow(WorkflowId.fromString(entry.subWorkflowExecutionUuid))
+    case Event(_: SubWorkflowNotFound, _) =>
+      prepareSubWorkflow(createSubWorkflowId())
+    case Event(SubWorkflowStoreFailure(command, reason), _) =>
+      jobLogger.error(reason, s"SubWorkflowStore failure for command $command, starting sub workflow with fresh ID.")
+      prepareSubWorkflow(createSubWorkflowId())
+  }
+  
+  when(SubWorkflowPreparingState) {
+    case Event(SubWorkflowPreparationSucceeded(subWorkflowEngineDescriptor, inputs), _) =>
+      startSubWorkflow(subWorkflowEngineDescriptor, inputs)
+    case Event(failure: CallPreparationFailed, _) =>
+      context.parent ! failure
+      context stop self
+      stay()
+  }
+  
+  when(SubWorkflowRunningState) {
+    case Event(WorkflowExecutionSucceededResponse(executedJobKeys, outputs), _) =>
+      context.parent ! SubWorkflowSucceededResponse(key, executedJobKeys, outputs)
+      goto(SubWorkflowSucceededState)
+    case Event(WorkflowExecutionFailedResponse(executedJobKeys, reason), _) =>
+      context.parent ! SubWorkflowFailedResponse(key, executedJobKeys, reason)
+      goto(SubWorkflowFailedState)
+    case Event(WorkflowExecutionAbortedResponse(executedJobKeys), _) =>
+      context.parent ! SubWorkflowAbortedResponse(key, executedJobKeys)
+      goto(SubWorkflowAbortedState)
+  }
+  
+  when(SubWorkflowSucceededState) { FSM.NullFunction }
+  when(SubWorkflowFailedState) { FSM.NullFunction }
+  when(SubWorkflowAbortedState) { FSM.NullFunction }
+  
+  whenUnhandled {
+    case Event(SubWorkflowStoreRegisterSuccess(command), _) =>
+      // Nothing to do here
+      stay()
+    case Event(SubWorkflowStoreCompleteSuccess(command), _) =>
+      // Nothing to do here
+      stay()
+    case Event(SubWorkflowStoreFailure(command, reason), _) =>
+      jobLogger.error(reason, s"SubWorkflowStore failure for command $command")
+      stay()
+    case Event(MetadataPutFailed(action, error), _) =>
+      jobLogger.warn(s"Put failed for Metadata action $action", error)
+      stay()
+    case Event(MetadataPutAcknowledgement(_), _) => stay()
+  }
+
+  onTransition {
+    case (fromState, toState) if fromState.workflowState != toState.workflowState =>
+      stateData.subWorkflowId foreach { id => pushCurrentStateToMetadataService(id, toState.workflowState) } 
+  }
+  
+  onTransition {
+    case (fromState, subWorkflowTerminalState: SubWorkflowTerminalState) =>
+      stateData.subWorkflowId match {
+        case Some(id) => pushWorkflowEnd(id)
+        case None => jobLogger.error("Sub workflow completed without a Sub Workflow UUID.")
```

Comment:
I don't think it can, but I figured that was better than a `stateData.subWorkflowId.get`